src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,446 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
 using Nethermind.Consensus.Messages;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
+using Nethermind.TxPool;
+using Nethermind.Core.Crypto;
 using Nethermind.Crypto;
 using Nethermind.Evm;
 using Nethermind.Int256;
-using Nethermind.TxPool;
 
-namespace Nethermind.Consensus.Validators
+namespace Nethermind.Consensus.Validators;
+
+public sealed class TxValidator : ITxValidator
 {
-    public class TxValidator : ITxValidator
+    private readonly ITxValidator[] _validators;
+
+    public TxValidator(ulong chainId)
     {
-        private readonly ulong _chainIdValue;
+        _validators = new ITxValidator[byte.MaxValue + 1];
+        _validators[(byte)TxType.Legacy] = new CompositeTxValidator([
+            IntrinsicGasTxValidator.Instance,
+            new LegacySignatureTxValidator(chainId),
+            ContractSizeTxValidator.Instance,
+            NonBlobFieldsTxValidator.Instance,
+        ]);
+        _validators[(byte)TxType.AccessList] = new CompositeTxValidator([
+            new ReleaseSpecTxValidator(static spec => spec.IsEip2930Enabled),
+            IntrinsicGasTxValidator.Instance,
+            SignatureTxValidator.Instance,
+            new ExpectedChainIdTxValidator(chainId),
+            ContractSizeTxValidator.Instance,
+            NonBlobFieldsTxValidator.Instance,
+        ]);
+        _validators[(byte)TxType.EIP1559] = new CompositeTxValidator([
+            new ReleaseSpecTxValidator(static spec => spec.IsEip1559Enabled),
+            IntrinsicGasTxValidator.Instance,
+            SignatureTxValidator.Instance,
+            new ExpectedChainIdTxValidator(chainId),
+            GasFieldsTxValidator.Instance,
+            ContractSizeTxValidator.Instance,
+            NonBlobFieldsTxValidator.Instance,
+        ]);
+        _validators[(byte)TxType.Blob] = new CompositeTxValidator([
+            new ReleaseSpecTxValidator(static spec => spec.IsEip4844Enabled),
+            IntrinsicGasTxValidator.Instance,
+            SignatureTxValidator.Instance,
+            new ExpectedChainIdTxValidator(chainId),
+            GasFieldsTxValidator.Instance,
+            ContractSizeTxValidator.Instance,
+            BlobFieldsTxValidator.Instance,
+            MempoolBlobTxValidator.Instance
+        ]);
+    }
 
-        public TxValidator(ulong chainId)
+    public TxValidator WithValidator(TxType type, ITxValidator validator)
```

Comment:
Having something like `WithValidator` mutate the object itself looks weird. I'm not sure if this is how we do it in other parts of the code. I know we do it for object builders (e.g. `TransactionBuilder`) but not for the object itself

Reply:
I would implement `WithValidator` as a clone, that is, return a new `TxValidator` with the added validator, but I thought that it made little sense to pay for that due to our use case. Do you have any specific suggestion?